src/Symfony/Bundle/FrameworkBundle/Debug/TraceableEventDispatcher.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\Debug;
+
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Console\ConsoleEvents;
+use Symfony\Component\Console\Event\ConsoleCommandEvent;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\HttpKernel\Debug\TraceableEventDispatcher as BaseTraceableEventDispatcher;
+use Symfony\Component\Stopwatch\Stopwatch;
+
+/**
+ * @internal
+ */
+final class TraceableEventDispatcher extends BaseTraceableEventDispatcher
+{
+    public function __construct(EventDispatcherInterface $dispatcher, Stopwatch $stopwatch, LoggerInterface $logger = null, private ?RequestStack $requestStack = null)
```

Comment:
let's move the CPP to a regular property, it's hard to spot currently

also wouldn't it make sense to type-hint for DebugRequestSTack instead? to be explicit about the coupling to this concept?

Reply:
I went for multi-lines constructor, as we should always do when using CPP.